src/codehelp/codeaction.vala

```diff
@@ -0,0 +1,477 @@
+/* codeaction.vala
+ *
+ * Copyright 2022 JCWasmx86 <JCWasmx86@t-online.de>
+ *
+ * This file is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+using Gee;
+using Lsp;
+
+namespace Vls.CodeActionExtractor {
```

Comment:
I want to organize this better so that the code is clearer to follow. For example, each type of code action could extend `Lsp.CodeAction`, with a constructor that builds itself from the code node or error.

Reply:
Should I add a directory like `codehelp/actions` and add a sourcefile for each action or should everything be in one big file?

Reply:
I think for now we can have everything in one file. As we add more code actions and as they become more complex, we may think of keeping them in separate files.